File: frontend/src/domain/subscription.ts
import type { MathSettings } from './learning'
import type { ReadingMode } from './reading'
export type LearningProfile = { birthDate: string; mathLevel: '' | MathSettings['level']; readingLevel: '' | ReadingMode }
export type LearningPolicy = { age?: number; mathLevel: MathSettings['level']; readingLevel: ReadingMode; version: number }
export type FamilySubscription = { familyId: number; name: string; registeredAt: string; status: string; plan: string; childLimit: number; activeChildren: number; accessUntil: string | null; paid: boolean; access: boolean }
export function allowsMath(policy: LearningPolicy, level: MathSettings['level']): boolean {
 const ranks = { easy: 1, medium: 2, hard: 3, columnar: 3 }
 return level === 'columnar' ? policy.mathLevel === 'columnar' : ranks[level] <= ranks[policy.mathLevel]
}
export function allowsReading(policy: LearningPolicy, level: ReadingMode): boolean {
 const ranks = { phrases: 1, sentences: 2, advanced: 3 }
 return ranks[level] <= ranks[policy.readingLevel]
}

File: backend/internal/domain/subscription.go
package domain

import "time"

type FamilySubscription struct {
	FamilyID       int64      `json:"familyId"`
	Name           string     `json:"name"`
	RegisteredAt   time.Time  `json:"registeredAt"`
	Status         string     `json:"status"`
	Plan           string     `json:"plan"`
	ChildLimit     int        `json:"childLimit"`
	ActiveChildren int        `json:"activeChildren"`
	AccessUntil    *time.Time `json:"accessUntil"`
	Paid           bool       `json:"paid"`
	Access         bool       `json:"access"`
	OwnerEmail     string     `json:"ownerEmail,omitempty"`
}
